### PR TITLE
minimum/maximum value of the all-positive/negative data

### PR DESCRIPTION
**Pull request by withmywoessner**
#### Reference issue
#12381

#### What does this implement/fix?
Currently, `evoked.get_peak()` can't find a min/max in all positive/ all negative data.
Implementation:
`evoked.get_peak(mode="neg", strict=False)
`

`strict `determines whether to return error if mode="neg" and there are no negative values (the current behavior, would become the strict=True behavior), or instead to return the minimum value of the all-positive data (the strict=False behavior).

### CODE REVIEW

File: mne/evoked.py [hunk 1]
```diff
@@ -1965,10 +1965,12 @@ def _get_peak(data, times, tmin=None, tmax=None, mode="abs"):
         The minimum point in time to be considered for peak getting.
     tmax : float | None
         The maximum point in time to be considered for peak getting.
-    mode : {'pos', 'neg', 'abs'}
+    mode : {'pos', 'neg', 'abs', 'min', 'max'}
```
Comment (drammock) [inline mne/evoked.py:1968]: confused; this doesn't match the API in the PR description (as was discussed in #12381)

Comment (withmywoessner) [reply mne/evoked.py:1968]: Sorry about that! I didn't push my changes yet.

**Comment by drammock** [comment]
I see that you've marked this as `draft` and I shouldn't have been so hasty to review. FYI: you can mark as draft *when first creating the PR* (the green "create pull request" button has a dropdown arrow that switches the button to "create draft pull request").  If you open a regular PR first and then later convert to draft, *a bunch of people are automatically subscribed to activity on this PR, and converting to draft mode later doesn't unsubscribe them*.  It's a quirk of how GitHub works and it often leads me to review work that isn't ready for review because it shows up in my inbox (which it wouldn't if the PR were made in draft mode initially).

**Comment by withmywoessner** [comment]
> I see that you've marked this as `draft` and I shouldn't have been so hasty to review. FYI: you can mark as draft _when first creating the PR_ (the green "create pull request" button has a dropdown arrow that switches the button to "create draft pull request"). If you open a regular PR first and then later convert to draft, _a bunch of people are automatically subscribed to activity on this PR, and converting to draft mode later doesn't unsubscribe them_. It's a quirk of how GitHub works and it often leads me to review work that isn't ready for review because it shows up in my inbox (which it wouldn't if the PR were made in draft mode initially).

Okay, thanks for letting me know! I'll be sure to do that next time.

File: mne/evoked.py [hunk 2]
```diff
@@ -914,6 +914,7 @@ def get_peak(
         time_as_index=False,
         merge_grads=False,
         return_amplitude=False,
+        strict=True,
```
Comment (larsoner) [inline mne/evoked.py:917]: We try to add these when not present when adding new kwargs these days
```suggestion
        *,
        strict=True,
```

File: mne/evoked.py [hunk 3]
```diff
@@ -2008,15 +2019,17 @@ def _get_peak(data, times, tmin=None, tmax=None, mode="abs"):
...
     elif mode == "neg":
-        if not np.any(data[~mask] < 0):
-            raise ValueError(
-                "No negative values encountered. Cannot " "operate in neg mode."
-            )
+        if strict:
+            if not np.any(data[~mask] < 0):
+                raise ValueError(
+                    "No negative values encountered. Cannot " "operate in neg mode."
+                )
```
Comment (larsoner) [inline mne/evoked.py:2032]:
```suggestion
        if strict and not np.any(data[~mask] < 0):
            raise ValueError(
                "No negative values encountered. Cannot " "operate in neg mode."
            )
```

**Comment by withmywoessner** [comment]
Done! @larsoner Thank you

**Review by drammock — APPROVED** [review]

**Comment by drammock** [comment]
@larsoner I let you merge if happy

**Comment by larsoner** [comment]
Thanks @withmywoessner !

**Pull request merged**